src/state/mod.rs

```diff
@@ -108,78 +114,52 @@ impl PersistedState {
             Actions::UpdateConversation(info) => {
-                let mut next = PersistedState {
-                    current_chat: self.current_chat,
-                    all_chats: self.all_chats.clone(),
-                    favorites: self.favorites.clone(),
-                    hide_sidebar: self.hide_sidebar,
-                    total_unreads: total_notifications(&self),
-                };
-                // overwrite the existing entry
-                next.all_chats.insert(info.conversation.id(), info);
-                next
+                state.all_chats.insert(info.conversation.id(), info);
+                state.total_unreads = total_notifications(&state);
+            }
+            Actions::UpdateFavorites(favorites) => {
+                state.favorites = favorites;
+            }
+            Actions::HideSidebar(slide_bar_bool) => {
+                state.hide_sidebar = slide_bar_bool;
             }
-            Actions::UpdateFavorites(favorites) => PersistedState {
-                current_chat: self.current_chat,
-                all_chats: self.all_chats.clone(),
-                favorites,
-                hide_sidebar: self.hide_sidebar,
-                total_unreads: self.total_unreads,
-            },
-            Actions::HideSidebar(slide_bar_bool) => PersistedState {
-                current_chat: self.current_chat,
-                all_chats: self.all_chats.clone(),
-                favorites: self.favorites.clone(),
-                hide_sidebar: slide_bar_bool,
-                total_unreads: self.total_unreads,
-            },
-            // Actions::SendNotification(title, content, sound) => {
-            //     let _ = PushNotification(title, content, sound);
-            //     PersistedState {
-            //         current_chat: self.current_chat,
-            //         all_chats: self.all_chats.clone(),
-            //         favorites: self.favorites.clone(),
-            //         hide_sidebar: self.hide_sidebar,
-            //         total_unreads: total_notifications(&self),
-            //     }
-            // }
+            Actions::SetShowPrerelaseNotice(value) => {
+                state.show_prerelease_notice = value;
+            } // Actions::SendNotification(title, content, sound) => {
+              //     let _ = PushNotification(title, content, sound);
+              //     PersistedState {
+              //         current_chat: self.current_chat,
+              //         all_chats: self.all_chats.clone(),
+              //         favorites: self.favorites.clone(),
+              //         hide_sidebar: self.hide_sidebar,
+              //         total_unreads: total_notifications(&self),
+              //     }
+              // }
         };
         // only save while there's a lock on PersistedState
-        next.save();
+        state.save();
 
         // modify PersistedState via assignment rather than mutation
-        *self = next;
+        *self = state;
```

Comment:
you should be able to do 

```rust
self.save();
```

instead of mutating `self` with the state itself.

Reply:
Does this commit fix it? 8a2fecb